docs/csharp/tutorials/pattern-matching.md

```diff
@@ -0,0 +1,325 @@
+---
+title: Use pattern matching features to extend data types
+description: This advanced tutorial demonstrates how to use pattern matching techniques to create functionality using data and algorithms that are created separately.
+ms.date: 03/13/2019
+ms.custom: mvc
+---
+# Tutorial: Using pattern matching features to extend data types
+
+C# 7 introduced basic pattern matching features. Those features are extended in C# 8 with new expressions and patterns. You can write functionality that behaves as though you extended types that may be in other libraries. Another use for patterns is to create functionality your application requires that isn't a fundamental feature of the type being extended.
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+> * How to recognize situations where pattern matching should be used.
+> * How to use pattern matching expressions to implement behavior based on types and property values.
+> * How to combine pattern matching with other techniques to create complete algorithms.
+
+## Prerequisites
+
+You'll need to set up your machine to run .NET Core, including the C# 8.0 preview compiler. The C# 8 preview compiler is available with [Visual Studio 2019 preview 4](https://visualstudio.microsoft.com/vs/preview/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019+preview), or [.NET Core 3.0 preview 3](https://dotnet.microsoft.com/download/dotnet-core/3.0).
```

Comment:
I would change the **preview 4** and **preview 3** to **latest preview**. Using **preview #** will become out of date very fast as the next preview or RC gets released.